api/types/app.go

```diff
@@ -296,9 +304,21 @@ func (a *AppV3) CheckAndSetDefaults() error {
 		}
 	}
 	if a.Spec.URI == "" {
-		return trace.BadParameter("app %q URI is empty", a.GetName())
+		if a.Spec.Cloud != "" {
+			a.Spec.URI = fmt.Sprintf("cloud://%v", a.Spec.Cloud)
```

Comment:
Why we need artificially set the URI if we have the a.spec.Cloud property  ?

Reply:
I'm pretty sure we make an assumption in other places in code that URI is never empty. This is a reasonable assumption to make, so instead of invalidating it we introduce a new scheme instead. We have an existing precedent for this approach: TCP apps which employ `tcp://...` scheme.